### PR TITLE
Enhancement: add signalhandling

### PR DESCRIPTION
**Pull request by igalic**
This pull request aims to add signal-handling to void

- [x] add nix as dependency
- [x] add `signals` to the `Config` structure
- [x] initialize `Config.signals` with various signals that bind to `Action::Quit`
- [ ] add `handle_signal()` akin to `handle_event()`
- [ ] call it in `run()`

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use nix::sys::signal::Signal;
+use nix::sys::signal::Signal::*;
```
Comment (spacekookie) [inline src/config.rs]: I'd prefer this import to be `nix::sys::signal{self, *};`

(sorry this is super nit-picky :sweat_smile::sweat_smile:)

**Comment by igalic** [comment]
I'm still weighing which signal library to use

nix seems extremely low level………

**Comment by spacekookie** [comment]
Yea, I noticed that too. https://github.com/kitsuneninetails/signal-rust looks pretty cool although could use some general love

**Comment by richardwesthaver** [comment]
if you're still looking to add sig handling may be worth checking out https://github.com/vorner/signal-hook

**Comment by igalic** [comment]
that's a good idea, but i have no time right now 😞

**Pull request closed without merging**